meetings/__init__.py

```diff
@@ -67,18 +75,37 @@ async def log_tag(self, tag, evt: MessageEvent) -> None:
             UPDATE meeting_logs SET tag = $3 WHERE meeting_id = $1 AND timestamp = $2
           """
     await self.database.execute(dbq, meeting, timestamp, tag)
+  
+  async def change_topic(self, topic, evt: MessageEvent) -> None:
+    dbq = """
+            UPDATE meetings SET topic = $3 WHERE meeting_id = $1 AND room_id = $2
+          """
+    await self.database.execute(dbq, self.meeting_id(evt.room_id), evt.room_id, topic)
 
-  # Helper: upload a JSON file
-  async def upload_file(self, evt, items, filename):
-    if len(items) == 0:
-      return
-    
-    logs = tuple()
-    for row in items:
-      log = f"{row[1]} <{row[2]}> #{row[3]}"
-      logs += (log,)
-    
-    data = "\n".join(logs).encode("utf-8")
+    # change the topic of the ^topic message to the topic
+    timestamp = evt.timestamp
+    dbq = """
+            UPDATE meeting_logs SET topic = $3 WHERE meeting_id = $1 AND timestamp = $2
+          """
+    await self.database.execute(dbq, self.meeting_id(evt.room_id), timestamp, topic)
+
+  async def log_to_db(self, meeting, timestamp, sender, message, topic):
+      # Log the item to the db
+      dbq = """
+              INSERT INTO meeting_logs (meeting_id, timestamp, sender, message, topic) VALUES ($1, $2, $3, $4, $5)
+            """
+      await self.database.execute(dbq, meeting, timestamp, sender, message, topic)
+
+  async def send_respond(self, event, message, meeting=None):
+      # could not figure out how to get maubot to work passively on itself, so manually log messages
+      # here instead
```

Comment:
Yeah, the `passive` thing is needed because of the IRC relay we're using (which prefixes line from IRC with `sender> ` - obviously that breaks commands because they need to be at the start of the line. So for IRC folk to log items, I had to use @passive.

Longer term I want to allow marking prior messages with :pencil2: or :triangular_flag_on_post: as well, so you can go back and say "oh, that should be added to the items" - which would also mean a Matrix user can manually flag IRC items too.